src/linux/mod.rs

```diff
@@ -25,23 +23,48 @@ pub struct LinuxSandbox {
     full_env: bool,
 }
 
-impl Sandbox for LinuxSandbox {
-    fn new() -> Result<Self> {
-        // Setup landlock filtering.
-        let mut landlock = Ruleset::new()
-            .set_best_effort(false)
-            .handle_access(AccessFs::from_all(ABI))?
-            .create()?;
-        landlock.as_mut().set_no_new_privs(true);
+impl LinuxSandbox {
+    /// Create a customized Linux sandbox.
+    ///
+    /// The [`min_landlock_abi`] argument defines the minimum Landlock Kernel
+    /// ABI version which must be supported. Sandboxing will fail on systems
+    /// which do not support this.
+    ///
+    /// All landlock ABI versions after [`min_landlock_abi`] versions are used
+    /// on systems that support them, but are ignored otherwise. This means
+    /// that the sandbox will be created without any error even if these are
+    /// not supported.
+    pub fn new_with_version(min_landlock_abi: LANDLOCK_ABI) -> Result<Self> {
+        let mut ruleset = Ruleset::new();
+
+        // Require at least `min_landlock_abi`.
+        (&mut ruleset).set_compatibility(CompatLevel::HardRequirement);
```

Comment:
Hmm, you pointed out to this change here: landlock-lsm/rust-landlock/issues/48

Reply:
Yes, otherwise `ruleset` would get moved since it returns whatever it is called on, which would be `Self` in this case (rather than `&mut Self`).

It would be cool if Rust would figure out automatically that it will work with just `&mut` access, but that might be a bit too much "magic".

Reply:
I prefer being able to split the methods apart for documentation purposes.

The second example makes things worse (imo). Though I've changed it to `ruleset = ruleset.set_compatibility` in this part of the code. It's just the `add_rule` part which uses `(&mut self.landlock)` now.